src/solver/opamActionGraph.ml

```diff
@@ -20,30 +20,36 @@ module type ACTION = sig
   type package
   module Pkg : GenericPackage with type t = package
   include OpamParallel.VERTEX with type t = package action
-  val to_aligned_strings: t list -> string list
+  val to_string: [< t ] -> string
+  val to_aligned_strings: [< t ] list -> string list
 end
 
-let action_strings ?utf8 x =
-  if utf8 = None && (OpamConsole.utf8 ()) || utf8 = Some true then
-    List.assoc x
-      [`inst,   "\xe2\x88\x97 ";
-       `up,     "\xe2\x86\x97 ";
-       `down,   "\xe2\x86\x98 ";
-       `reinst, "\xe2\x86\xbb ";
-       `rm,     "\xe2\x8a\x98 "]
-  else
-    List.assoc x
-      [`inst,   "install";
-       `up,     "upgrade";
-       `down,   "downgrade";
-       `reinst, "recompile";
-       `rm,     "remove"]
+let name_of_action = function
+  | `Remove _ -> "remove"
+  | `Install _ -> "install"
+  | `Change (`Up,_,_) -> "upgrade"
+  | `Change (`Down,_,_) -> "downgrade"
+  | `Reinstall _ -> "recompile"
+  | `Build _ -> "build"
+
+let symbol_of_action = function
+  | `Remove _ -> "\xe2\x8a\x98 "
```

Comment:
Just a comment. I think it's nice when you write verbatim UTF-8 to indicate in a comment to which code point this corresponds to using the `U+XXXX` notation.